apps/hash-graph/lib/graph/src/api/rest/entity_type.rs

```diff
@@ -105,12 +129,70 @@ async fn create_entity_type<P: StorePool + Send>(
     body: Json<CreateEntityTypeRequest>,
     // TODO: We want to be able to return `Status` here we should try and create a general way to
     //  call `status_to_response` for our routes that return Status
-) -> Result<Json<ListOrValue<OntologyElementMetadata>>, Response> {
-    let Json(CreateEntityTypeRequest {
+) -> Result<Json<ListOrValue<OntologyElementMetadata>>, Response>
+where
+    for<'pool> P::Store<'pool>: RestApiStore,
+{
+    let mut store = pool.acquire().await.map_err(|report| {
+        tracing::error!(error=?report, "Could not acquire store");
+        status_to_response(Status::new(
+            hash_status::StatusCode::Internal,
+            Some(
+                "Could not acquire store. This is an internal error, please report to the \
+                 developers of the HASH Graph with whatever information you can provide including \
+                 request details and logs."
+                    .to_owned(),
+            ),
+            vec![StatusPayloads::ErrorInfo(ErrorInfo::new(
+                // TODO: add information from the report here
+                //   https://app.asana.com/0/1203363157432094/1203639884730779/f
+                HashMap::new(),
+                // TODO: We should encapsulate these Reasons within the type system, perhaps
+                //  requiring top level contexts to implement a trait `ErrorReason::to_reason`
+                //  or perhaps as a big enum, or as an attachment
+                "STORE_ACQUISITION_FAILURE".to_owned(),
+            ))],
+        ))
+    })?;
+
+    #[allow(clippy::infallible_destructuring_match)]
+    let CreateOwnedEntityTypeRequest {
         schema,
         owned_by_id,
         actor_id,
-    }) = body;
+    } = match body.0 {
+        CreateEntityTypeRequest::Owned(request) => request,
+        #[cfg(feature = "type-fetcher")]
+        CreateEntityTypeRequest::External(request) => {
+            return Ok(Json(ListOrValue::Value(
+                store
+                    .load_external_type(
+                        &domain_validator,
+                        OntologyTypeReference::DataTypeReference((&request.schema).into()),
```

Comment:
```suggestion
                        OntologyTypeReference::EntityTypeReference((&request.schema).into()),
```
Would this have resulted in errors in other places?

Reply:
Thanks for finding this, this wasn't intentional. It didn't break anything because I only used the `.url()` method.